proto/client.proto

```diff
@@ -51,7 +51,14 @@ message UpdateRequest {
 	Profile profile = 2	[(gogoproto.customtype) = "EncodedProfile", (gogoproto.nullable) = false];
 	LookupRequest lookup_parameters = 3;
 
-	bytes dkim_proof = 1000 [(gogoproto.customname) = "DKIMProof"];
+	oneof email_proof {
+		bytes dkim_proof = 1000 [(gogoproto.customname) = "DKIMProof"];
+		// oidc_token contains OpenID Connect token received from the provider
+		string oidc_token = 1001 [(gogoproto.customname) = "OIDCToken"];
+		// auth_data is an opaque field and is primarily provided in case keyserver
+		// wants to do custom verification to confirm ownership of an email address
+		bytes auth_data = 1002;
```

Comment:
I don't think we should declare a field without documenting its contents. How about `// new name ownership proof types can be assigned to fields 1002 through 1100; please make your best effort not to use a number that is already used in a different way by somebody else`. A message containing an unknown `email_proof` will be interpreted as if it had no email proof by the protobuf machinery, which I think is appropriate. The way I understand protobufs is that all extensibility should be of the form "to add a new option, add a new field".